### PR TITLE
Ensure use_mv_v2_evaluation is passed through to feature state evaluation

### PR DESCRIPTION
**Pull request by matthewelwell**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

Ensures that the environment `use_mv_v2_evaluation` attribute is passed through to the retrieval of the flag value in the Core API code path. 

## How did you test this code?

Added new unit test and updated existing.

### CODE REVIEW

File: api/features/models.py [hunk 1]
```diff
@@ -603,7 +603,13 @@ def get_feature_state_value_by_hash_key(
         return feature_state_value and feature_state_value.value
 
     def get_feature_state_value(self, identity: "Identity" = None) -> typing.Any:
-        identity_hash_key = identity.get_hash_key() if identity else None
+        identity_hash_key = (
+            identity.get_hash_key(
+                use_mv_v2_evaluation=identity.environment.use_mv_v2_evaluation
```
Comment (khvn26) [inline api/features/models.py:608]: I wonder if we could default to `identity.environment.use_mv_v2_evaluation` inside engine's `Identity.get_hash_key`.

File: api/tests/unit/integrations/amplitude/test_unit_amplitude.py [hunk 2]
```diff
@@ -25,33 +23,30 @@ def test_amplitude_initialized_correctly():
 
 
 @pytest.mark.django_db
-def test_amplitude_when_generate_user_data_with_correct_values_then_success():
+def test_amplitude_when_generate_user_data_with_correct_values_then_success(
+    environment: Environment,
+    feature_state: FeatureState,
+    feature_state_with_value: FeatureState,
+    identity: Identity,
+):
```
Comment (khvn26) [inline api/tests/unit/integrations/amplitude/test_unit_amplitude.py]:
```suggestion
) -> None:
```

File: api/tests/unit/integrations/heap/test_unit_heap.py [hunk 3]
```diff
@@ -2,37 +2,38 @@
...
 
 
 @pytest.mark.django_db
-def test_heap_when_generate_user_data_with_correct_values_then_success():
+def test_heap_when_generate_user_data_with_correct_values_then_success(
+    environment: Environment,
+    feature_state: FeatureState,
+    feature_state_with_value: FeatureState,
+    identity: Identity,
+):
```
Comment (khvn26) [inline api/tests/unit/integrations/heap/test_unit_heap.py]:
```suggestion
) -> None:
```

File: api/tests/unit/integrations/segment/test_unit_segment.py [hunk 4]
```diff
@@ -22,31 +20,29 @@ def test_segment_initialized_correctly():
 
 
 @pytest.mark.django_db
-def test_segment_when_generate_user_data_with_correct_values_then_success():
+def test_segment_when_generate_user_data_with_correct_values_then_success(
+    environment: Environment,
+    feature_state: FeatureState,
+    feature_state_with_value: FeatureState,
+    identity: Identity,
+):
```
Comment (khvn26) [inline api/tests/unit/integrations/segment/test_unit_segment.py]:
```suggestion
) -> None:
```

File: api/tests/unit/integrations/heap/test_unit_heap.py [hunk 5]
```diff
@@ -2,37 +2,38 @@
...
 
     # Then
     expected_user_data = {
         "app_id": api_key,
         "identity": identity.identifier,
         "event": "Flagsmith Feature Flags",
-        "properties": {"Test Feature": False},
+        "properties": {
+            feature_state.feature.name: feature_state.enabled,
+            feature_state_with_value.feature.name: feature_state_with_value.get_feature_state_value(
```
Comment (khvn26) [inline api/tests/unit/integrations/heap/test_unit_heap.py]: I'd prefer a hard-coded value here.

File: api/features/models.py (hunk 1 above)
Comment (matthewelwell) [reply api/features/models.py:608]: So, this was discussed on the PR for the original implementation here: https://github.com/Flagsmith/flagsmith/pull/1885#discussion_r1095621725. 

I am leaning towards removing the method argument and reverting to use the value dynamically from the identity though.

**Comment by luk355** [comment]
Thanks for fixing this, @matthewelwell. I can confirm the multivariate FFs now work consistently in UI, API and .NET local evaluation mode [in my fork](https://github.com/luk355/flagsmith-dotnet-client/tree/lb/chore/cleanup).

File: api/features/models.py (hunk 1 above)
Comment (khvn26) [reply api/features/models.py:608]: So am I, unless we have a clear use case for the current API — e.g. an API configuration setting.

**Review by khvn26 — APPROVED** [review]

**Pull request merged**